README.md

```diff
@@ -347,9 +378,77 @@ knows Staltz is part of the group because all `group/add-member` messages have
 Staltz can see Arj has invited Mix because he's replicating Arj's
 "group/additions" feed, so Staltz starts replicating Mix's group feed.
 
+## 5. Security Considerations
+
+### 5.1. Degraded confidentiality at scale
+
+As noted in the [private-groups-original-notes], SSB Private Groups are based on
+symmetric encryption and static keys (no key rotation), which means it cannot
+guarantee [perfect-forward-secrecy] neither [post-compromise-security]. This
+makes SSB Private Groups less likely to guarantee confidentiality at scale, when
+a group has dozens or hundreds of members. If any of the members leaks the group
+secret, all past communication in the group is compromised.
+
+This also means that any group member can add new members to the group. They may
+even do so without publishing a `group/add-member` message, which effectively
+allows any group member to invite a third party that will remain undetected by
+existing group members.
+
+SSB Private Groups with several members should thus not be considered
+confidential nor forward secret, and should be used with caution. Our security
+model assumes that the group is as trustworthy as the _least_ trustworthy member
+in the group.
+
+The choice of symmetric encryption (as opposed to, e.g. hash-ratchet or
+double-ratchet) was made because of efficiency. Symmetric encryption means
+adding new members or publishing new content is `O(1)` fast, and the system is
+overall simple to implement.
+
+### 5.2. Membership analysis attacks
+
+A group feed is discoverable only by the group members, because its announcement
+is encrypted. However, an eavesdropper who can replicate a peer's metafeed tree
+will notice the presence of these encrypted message on shard feeds. If the
+eavesdropper can do that for several peers, then they may perform a metadata
+analysis attack where they try to correlate the encrypted messages on the shard
+feeds to the group members.
+
+This can be mitigated by publishing dummy encrypted messages on the shard feed
+at random intervals, with the downside of increasing the size of the shard feed,
+and thus making partial replication heavier.
+
+It is currently unknown how much can be learned from such analysis attack, as
+it depends on the network's complexity, such as number of peers, number of
+groups, and number of relationships between peers.
+
```

Comment:
```suggestion

Another way membership information can leak is through replication requests -
if only Alice, Bob, and Carol ask for feeds A, B, C, and these feeds don't seem to
be associated with any publicly declared sub-feeds, it might be reasonable to guess
that Alice, Bob, Carol are all in a group and that A, B, C are their associated feeds
for that group. We think this can be mitigated through "sympathetic replication"
(asking friends who aren't in our groups to replicate some of our group feeds) in order
to add more noise (and replication resilience) to the system. This needs further
investigation and specification.
```

Reply:
Good point